### PR TITLE
Added option to install 32 bit version of Node.js on x64 agents

### PR DESCRIPTION
**Pull request by teilmeier**
**Task name**: NodeToolV0 and UseNodeV1

**Description**: Added checkbox input that allows users to choose installation of 32 bit Node.js on the agent even if the agent has a x64 OS architecture. This is helpful if the software project needs the 32 bit version but also wants to leverage Cloud-hosted agents. Adjusted caching logic and architecture determination logic. Added translations (Microsoft Translator). 

**Documentation changes required:** Y

**Added unit tests:** N

**Attached related issue:** N

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Comment by nschonni** [comment]
Not sure if #8688 should get closed if this lands. This probably covers my original use case, but I went with other arches as a dropdown

**Comment by kenr** [comment]
I'd like to use this task to setup x86 version on a x64 machine on our self hosted agents. Is there anything that blocks this PR?

**Review by aminya — APPROVED** [review]

File: Tasks/NodeToolV0/task.json
Comment (damccorm) [inline Tasks/NodeToolV0/task.json:17]: Sorry I missed this when it came in! This looks like a good change to me - would you mind bumping the version here and in the task.loc.json to 0.176.0? Once that happens I'll go ahead and merge it

Comment (kenr) [reply Tasks/NodeToolV0/task.json:17]: Anything I can do to get this through?

File: Tasks/NodeToolV0/task.json
Comment (damccorm) [inline Tasks/NodeToolV0/task.json:16]:
```suggestion
        "Minor": 176,
```

File: Tasks/NodeToolV0/task.loc.json
Comment (damccorm) [inline Tasks/NodeToolV0/task.loc.json:16]:
```suggestion
    "Minor": 176,
```

**Review by damccorm — APPROVED** [review]

**Review by Close2Angelz — COMMENTED** [review]
Cross platform iOS variable

**Pull request merged**
